%YAML:1.0

#camera calibration 
model_type: PINHOLE
num_pairs: 2
camera_name: leftcamera
image_width: 752
image_height: 480



#feature traker paprameters
max_cnt: 100           # max feature number in feature tracking
min_dist: 50            # min distance between two features 
F_threshold: 1        # ransac threshold (pixel)
show_track: 1          # publish tracking image as topic
equalize: 0             # if image is too dark or light, trun on equalize to find enough features
fisheye: 0              # if using fisheye, trun on it. A circle mask will be loaded to remove edge noisy points

#================================ ROS Settings ================================
imu_topic: "/ardrone/vi_sensor_1/imu"
gps_topic: "/at_drone_interface/gps"
baro_topic: "/at_drone_interface/barometric_altitude"
mag_topic: "/at_drone_interface/magnetometer"
disp_topic: "/narrow_stereo/disparity"

left_img_topic: "/ardrone/vi_sensor_2/left/image_raw"
right_img_topic: "/ardrone/vi_sensor_2/right/image_raw"
odom_frame_id: "/world"
imu_frame_id: "/imu"

#================================= Frontend Settings ===========================
T_cam0_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data:  [0.0, -1.0, 0.0, 0.055,
          0.0, 0.0, -1.0, -0.0048,
          1.0, 0.0, 0.0, 0.0,
          0.0, 0.0, 0.0, 1.0]

cam0_distortion_coeffs: !!opencv-matrix    
  rows: 1
  cols: 5
  dt: d
  data: [0,0,0,0,0]


cam0_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [448.1008985853343, 448.1008985853343, 376.5, 240.5]


T_cam1_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data:  [0.0, -1.0, 0.0, -0.055,
          0.0, 0.0, -1.0, -0.0048,
          1.0, 0.0, 0.0, 0.0,
          0.0, 0.0, 0.0, 1.0]

cam1_distortion_coeffs: !!opencv-matrix   
  rows: 1
  cols: 5
  dt: d
  data: [0,0,0,0,0]

cam1_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [448.1008985853343, 448.1008985853343, 376.5, 240.5]


T_cam2_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data:  [0.0, 1.0,  0.0, 0.055,
          0.0,  0.0, -1.0, -0.0048,
          -1.0, 0.0, 0.0, -0.230,
          0.0, 0.0, 0.0, 1.0]

# T_cam2_imu: !!opencv-matrix
#   rows: 4
#   cols: 4
#   dt: d
#   data:  [0.0, 0.0,  -1.0, 0.055,
#           1.0,  0.0, 0.0, -0.0048,
#           0.0, -1.0, 0.0, -0.230,
#           0.0, 0.0, 0.0, 1.0]

cam2_distortion_coeffs: !!opencv-matrix    
  rows: 1
  cols: 5
  dt: d
  data: [0,0,0,0,0]


cam2_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [448.1008985853343, 448.1008985853343, 376.5, 240.5]

T_cam3_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data:  [0.0, 1.0,  0.0, -0.055,
          -0.0,  0.0, -1.0, -0.0048,
          -1.0, -0.0, -0.0, -0.230,
          0.0, 0.0, 0.0, 1.0]

# T_cam3_imu: !!opencv-matrix
#   rows: 4
#   cols: 4
#   dt: d
#   data:  [0.0, 0.0,  -1.0, -0.055,
#           1.0,  0.0, 0.0, -0.0048,
#           0.0, -1.0, 0.0, -0.230,
#           0.0, 0.0, 0.0, 1.0]


cam3_distortion_coeffs: !!opencv-matrix   
  rows: 1
  cols: 5
  dt: d
  data: [0,0,0,0,0]

cam3_intrinsics: !!opencv-matrix
  rows: 1
  cols: 4
  dt: d
  data: [448.1008985853343, 448.1008985853343, 376.5, 240.5]



T_W_imu: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  data: [1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1]
  
#================================== Estimator Settings ==========================
# 1 = standard projection factor, 2 = smartfactor
estimator_type: 2
publish_freq: 200
propagateAtImuRate: 0
fixlag: 1
dense: 0
sim: 0
lag_size: 7
use_barometer: 0
use_gps: 0

max_iter: 5
abs_tol: 1.0e-3
rel_tol: 1.0e-3
keyframe_spacing: 0.08

#================================== IMU Settings ==============================
# sigma_a, sigma_g, sigma_ba, sigma_bg
sigma_a: 2.0e-2
sigma_g: 1.6968e-3
sigma_ba: 3.0e-2
sigma_bg: 1.9393e-3
sigma_pixel: 1
#sigma: [8.0000e-3, 12.0e-4, 4.0e-5 , 4.0e-6 ]
#sigma: [2.0000e-4, 1.6968e-06 , 3.0000e-6 , 1.9393e-06]
preint_cov: 1.0e-3
preint_bias_cov: 1.0e-5

#================================== Debug Settings ==========================
show_landmarks: 1
logging: 1
offline: 0
output_path: "/home/joshua/traj_data/trajectory_temp_sim.csv"

# To take features from a single pair, specify the pair number. If negative, backend will operate in multi-camera mode
add_features: -1
